store/src/java/com/zimbra/cs/imap/ImapSearch.java

```diff
@@ -167,7 +172,16 @@ static class AndOperation extends LogicalOperation {
         @Override
         protected String toZimbraSearch(ImapFolder i4folder) throws ImapParseException {
             StringBuilder search = new StringBuilder("(");
+            Pattern compiledUidPattern = Pattern.compile(UID_SEARCH_PATTERN);
             for (ImapSearch i4search : mChildren) {
+                if (i4search instanceof SequenceSearch
+                        && isUidRangeSearchIgnored
+                        && null != ((SequenceSearch) i4search).mSubSequence) {
+                    Matcher uidMatcher = compiledUidPattern.matcher(((SequenceSearch) i4search).mSubSequence);
```

Comment:
there is no use of `uidMatcher` further can combine this with if check below `compiledUidPattern.matcher(((SequenceSearch) i4search).mSubSequence).find()`

Reply:
Fixed.